Add overview arch (#2209)
giantswarm/docs: src/content/overview/architecture/_index.md

  - What is the architecture of the Giant Swarm cloud-native developer platform?
owner:
  - https://github.com/orgs/giantswarm/teams/team-planeteers
last_review_date: 2024-04-26
last_review_date: 2024-11-04
---

> ## Platform API {#platform-api}

Author: @JosephSalisbury, I added the diagram. I added an intro, I explained the layers; and now my idea is to dissect all the boxes. I want to add a paragraph by box to avoid making it super long and point to the right section to read more. WDYT?

## Platform architecture

![Platform architecture](./platform-architecture.png)

The platform architecture consists of the following layers:

> - Infrastructure layer: responsible for managing the underlying infrastructure, including the cloud provider resources or on-premises infrastructure. To abstract the underlying infrastructure, the platform uses the `Kubernetes`.

Reviewer: i'd probably reverse these three, so they match the diagram

- Capabilities layer: provides a set of capabilities to manage the lifecycle of applications, including deploying, scaling, and monitoring applications.

- Interface layer: offers a set of templates, APIs, and tools that developers can use to interact with the platform.

Along these layers, Giant Swarm provides a real support channel, incident management, and good documentation to help you get the most out of the platform. Additionally, you have a dedicated account engineer to help creating a common roadmap and to provide guidance on how to use the platform.

## Platform API

The Giant Swarm platform is build on top of `Kubernetes` thanks to [`Cluster API`](https://cluster-api.sigs.k8s.io/), a open source  project that standardize the cluster lifecycle management across different cloud providers or on-premises infrastructure.

> In `Cluster API`, there is a distinction between the management cluster and the workload clusters. The management cluster is responsible for managing the lifecycle of the workload clusters. The workload clusters are the clusters where you deploy your applications.

Reviewer: I'm not sure if i'd go into MC/WC topology here somehow, we don't mention it in the diagram

Author: Hmm, I wanted to explain what the platform really is and the platform API. It is hard without introducing the concept of MC/WC. At the same time, all installation flavours have an MC/WC concept, though they could vary in form. I think we could go deeper in child documents explaining the different setups but introduce here the concept IMO is valid :D

Reviewer 2: Maybe don't go into details that much, but still keep the sentence:

```suggestion
The Platform API is the regular Kubernetes API of your central _management cluster_ and is your interface for deploying workload clusters and applications, or to reach other capabilities such as the observability and security dashboards.
```

## Platform architecture

![Platform architecture](./platform-architecture.png)

The platform architecture consists of the following layers:

- Interfaces: how administrators, developers and automation can access the platform, for example to deploy applications, monitor health of clusters and workloads or use other capabilities.

- Capabilities: features offered by the platform, such as application deployment, autoscaling, security or observability.

- Infrastructure: the Kubernetes-based platform can manage clusters on several cloud providers. We support AWS, Azure, VMware (on-premises) or a mixture thereof ("hybrid").

Along these layers, Giant Swarm provides a live support channel, incident management, and good documentation to help you get the most out of the platform. Additionally, you have a dedicated account engineer to help create a common roadmap and to provide guidance on how to use the platform for your goals.

## Platform API

> The Giant Swarm platform is build on top of `Kubernetes` based on [`Cluster API`](https://cluster-api.sigs.k8s.io/), an open source  sub-project that standardizes the cluster lifecycle management across different cloud providers or on-premises infrastructure.

Reviewer 2: All the backtick-quoted terms still need to be un-backticked, for example
```suggestion
The Giant Swarm platform is build on top of Kubernetes based on [Cluster API](https://cluster-api.sigs.k8s.io/), an open source Kubernetes sub-project that standardizes the cluster lifecycle management across different cloud providers or on-premises infrastructure.
```

> The platform API is the regular `Kubernetes` API of your central _management cluster_ and is your interface for deploying workload clusters and applications, or to reach other capabilities such as the observability and security dashboards. Thanks to being based on `Kubernetes`, you can use the same tools and workflows you are already familiar with, such as `kubectl`, `GitOps` or any other existing tooling compatible.

Reviewer 3: ```suggestion
The platform API is the regular `Kubernetes` API of your central _management cluster_ and is your interface for deploying workload clusters and applications, or to reach other capabilities such as the observability and security dashboards. Thanks to being based on `Kubernetes`, you can use the same tools and workflows you are already familiar with, such as `kubectl`, `GitOps` or any other existing Kubernetes-compatible tooling.
```

In Giant Swarm, the management cluster is also used for enhancing the platform with capabilities, such as monitoring, alerting, policy management, and many more. You can use the platform API of our management clusters to configure these capabilities and even add your own capabilities through operators.

[Learn how to access the platform API]({{< relref "/getting-started/access-to-platform-api" >}}).

## GitOps

> Giant Swarm uses GitOps as first-class citizen to manage the lifecycle of cluster and applications. In the platform, the `Flux` operator runs to ensure all changes are version-controlled, auditable, and easily reversible. You can use it too to manage your applications and configurations. It's as simple as adding your existing repositories to the platform and let the platform do the rest.

Reviewer 3: ```suggestion
Giant Swarm uses GitOps as first-class citizen to manage the lifecycle of cluster and applications. In the platform, the `Flux` operator runs to ensure all changes are version-controlled, auditable, and easily reversible. You can use it to manage your applications and configurations. It's as simple as adding your existing repositories to the platform and let the platform do the rest.
```

[Learn more in our guide about GitOps]({{< relref "/tutorials/continuous-deployment" >}}).

## Developer portal

Giant Swarm uses [Backstage](https://www.cncf.io/projects/backstage/) as developer portal serving as the central hub for accessing all platform services and resources. It provides an intuitive interface where platform engineers and developers find documentation, manage their projects, and access tools necessary for application development and deployment. By consolidating resources and simplifying access, your developer portal enhances the user experience, reduces onboarding time, and fosters collaboration among teams. It acts as the gateway to the platform’s capabilities, streamlining workflows and promoting productivity.

## Observability

> Observability on the Giant Swarm platform is based on the Grafana stack. Collectors store metrics, logs and traces in central storage and you can use them in the managed Grafana instance on your management cluster to get a centralized view, set up alerts and dashboards, or troubleshoot issues. In our tutorial (TODO LINK HERE OR BELOW), you see how easily your developers can start monitoring applications with this stack. Giant Swarm's pre-installed dashboards let you view the infrastructure and cluster health without extra effort.

Reviewer 3: don't forget your TODO here

## Single sign-on

Single Sign-On (SSO) simplifies user authentication across your use of the Giant Swarm platform by authenticating user access with your existing identity provider (for example `Google Workspaces` or `Microsoft Entra` / `Active Directory`). This enhances security and user convenience by centralizing identity management and not requiring a separate password. You can build on top of Kubernetes Role-Based Access Control (RBAC) to define fine-grained access policies and permissions, ensuring which users are authorized to access certain platform resources.

> TODO LINK OVERVIEW ARTICLE

Reviewer 3: see above

## Network and connectivity

> The different technologies used in the Giant Swarm platform provide you with a secure and reliable network and connectivity between your applications and services. Platform engineers can create network policies to control traffic flows and enforce security rules, ensuring that data is protected and isolated. Relying on the built-in networking capabilities of `Kubernetes` as well as the extended capabilities of `Cilium CNI`, the platform supports many routing and load balancing strategies, enabling seamless communication between services and applications.

Reviewer 3: This is still missing ingress

## Platform security

> Platform security is a critical aspect on the Giant Swarm platform, encompassing various components to ensure the safety and integrity of systems. [Kyverno](https://www.cncf.io/projects/kyverno/) plays a pivotal role as a policy engine by establishing, enforcing, and automating security policies across all your environments. These policies govern access controls, resource configurations, and compliance requirements, ensuring that all your workloads adhere to organizational and regulatory standards. On top of that, the platform provides intrusion detection, vulnerability scanning, and automated threat response capabilities thanks to [Falco](https://www.cncf.io/projects/falco/).

Reviewer 3: CVE scanning is not Falco but Trivy

  - What is the architecture of the Giant Swarm cloud-native developer platform?
owner:
  - https://github.com/orgs/giantswarm/teams/team-planeteers
last_review_date: 2024-11-04
---

> Giant Swarm's cloud-native platform is a collection of open-source components that work together to provide a seamless experience for managing the lifecycle of containerized applications. The platform is designed to be cloud-agnostic, allowing you to deploy your applications on any of the supported cloud providers. Besides, the platform offers a rich set of APIs that our customers can use to offer a set of self-service capabilities to their developers.

Reviewer 2: ```suggestion
Giant Swarm's cloud-native developer platform integrates open-source components that work together to provide a seamless experience for managing the lifecycle of containerized applications. The platform is based on Kubernetes and designed to be cloud-agnostic, allowing you to deploy your applications on any of the supported cloud providers, including on-premises. The interfaces allow development teams and automations to deploy applications, keep the clusters secure, and use other capabilities that we explain below. On top, you benefit from our strong support model that ensures your long-term success and stability of workloads.
```

Changes I made:

- Don't use the word "collection" since it sounds too much like the platform isn't adding much value and just puts pieces together
- Replaced "Rich set of APIs" since that's not a selling point. Less is more. We should rather keep it short here, and explain all interfaces below.
- Added a short note of the support model, just to have mentioned each part of the diagram also in the text.

## Platform architecture

![Platform architecture](./platform-architecture.png)

The platform architecture consists of the following layers:

> - Interface layer: offers a set of templates, APIs, and tools that developers can use to interact with the platform.

Reviewer 4: What about Happa and Backstage. Our CLI (kubectl extension) is an interface. Also GitOps via flux is an interface that is probably different enough to mention besides API.

Reviewer 2: IMO these bullet points should quickly summarize the diagram, and all details should go into separate sections per layer

> - Interface layer: offers a set of templates, APIs, and tools that developers can use to interact with the platform.

Reviewer 2: ```suggestion
- Interfaces: how administrators, developers and automation can access the platform, for example to deploy applications, monitor health of clusters and workloads or use other capabilities.
```

> - Capabilities layer: provides a set of capabilities to manage the lifecycle of applications, including deploying, scaling, and monitoring applications.

Reviewer 2: ```suggestion
- Capabilities: features offered by the platform, such as application deployment, auto-scaling, security or observability.
```

> - Infrastructure layer: responsible for managing the underlying infrastructure, including the cloud provider resources or on-premises infrastructure. To abstract the underlying infrastructure, the platform uses the `Kubernetes`.

Reviewer 3: ```suggestion
- Infrastructure layer: responsible for managing the underlying infrastructure, including the cloud provider resources or on-premises infrastructure. To abstract the underlying infrastructure, the platform uses `Kubernetes`.
```

> - Infrastructure layer: responsible for managing the underlying infrastructure, including the cloud provider resources or on-premises infrastructure. To abstract the underlying infrastructure, the platform uses the `Kubernetes`.

Reviewer 2: ```suggestion
- Infrastructure: the Kubernetes-based platform can manage clusters on several cloud providers. We support AWS, Azure, VMware (on-premises) or a mixture thereof ("hybrid").
```

> Along these layers, Giant Swarm provides a real support channel, incident management, and good documentation to help you get the most out of the platform. Additionally, you have a dedicated account engineer to help creating a common roadmap and to provide guidance on how to use the platform.

Reviewer 3: ```suggestion
Along these layers, Giant Swarm provides a live support channel, incident management, and good documentation to help you get the most out of the platform. Additionally, you have a dedicated account engineer to help create a common roadmap and to provide guidance on how to use the platform for your goals.
```

real sounds weird, what would the opposite be "fake"?

Reviewer 2: +1 for "live", or even 24/7 since together with incident handling, the "channel" is basically all day and night

## Platform API

> The Giant Swarm platform is build on top of `Kubernetes` thanks to [`Cluster API`](https://cluster-api.sigs.k8s.io/), a open source  project that standardize the cluster lifecycle management across different cloud providers or on-premises infrastructure.

Reviewer 3: ```suggestion
The Giant Swarm platform is build on top of `Kubernetes` based on [`Cluster API`](https://cluster-api.sigs.k8s.io/), an open source project that standardizes the cluster lifecycle management across different cloud providers or on-premises infrastructure.
```

Reviewer 4: I would start by mentioning that the platform API that is exposed is the kubernetes api. That all the exising tooling works with our platform api. Examples: GitOps, kubectl even ci/cd systems/terraform etc. Also the authentication and authorization is the same.

And then go into the topology. There we also need to mention that the management cluster (and the platform api) is not a saas by giantswarm. But instead deployed, maintained and operated by giant swarm within the premises of the customer.

Reviewer 2: +1. And: please don't quote Kubernetes or other terms (CAPI, Git, ...) as code block with backticks. Those terms are well-known and don't need formatting (no italics, no quoting, just a link and spelling out the abbreviation at first use is enough). We should mention CAPI as _subproject_ of Kubernetes to denote it as close neighbor.

In `Cluster API`, there is a distinction between the management cluster and the workload clusters. The management cluster is responsible for managing the lifecycle of the workload clusters. The workload clusters are the clusters where you deploy your applications.

> In Giant Swarm, the management cluster is also used for enhancing the platform capabilities, such as monitoring, logging, and alerting. The customer platform team uses the management cluster to configure which capabilities are available to the developers, and even to create new capabilities.

Reviewer 3: ```suggestion
In Giant Swarm, the management cluster is also used for enhancing the platform with capabilities, such as monitoring, logging, and alerting. The customer platform team uses the management cluster to configure which capabilities are available to the developers, and even to create new capabilities.
```

Reviewer 4: Looks like we jump back and forth between "the customer" and "you".  I'd suggest: 

We extend the management cluster concept for the platform with even more capabilities, such as monitoring, alerting, policy management, and many more. You can use the Platform API of our management clusters to configure these capabilities and even add your own capabilities through operators.

Reviewer 2: Yeah, let's please avoid "the customer" (and also "we" – should be "Giant Swarm").

I think we can keep it short in the platform overview and get rid of this paragraph.

> Most often, you have a single management cluster that manages multiple workload clusters, but in case of running infrastructure in multiple regions, you can have multiple management clusters. The platform API isn't more than the `Kubernetes` management cluster API, but with some additional resources that allow you to manage the lifecycle of your workloads across multiple clusters or inspect the application metrics of your fleet of clusters.

Reviewer 3: ```suggestion
Most often, you have a single management cluster that manages multiple workload clusters, but in case of running infrastructure in multiple regions, you can have multiple management clusters. The platform API is an extended `Kubernetes` API, comprising upstream APIs, like Cluster API and Flux, and some some additional resources that allow you to manage the lifecycle and security of your workloads across multiple clusters or inspect the application metrics of your fleet of clusters.
```

I think we should not downplay our API too much, it is much more powerful than anything you could get ootb

Reviewer 2: I'd remove this whole paragraph. It's about a specific requirement (multi-region) that doesn't belong in the overview. This also creates a pseudo recommendation of having multiple management clusters, which isn't a must.

[Learn how to access the platform API]({{< relref "/getting-started/access-to-platform-api" >}}).

## GitOps

> Giant Swarm uses GitOps as first-class citizen to manage the lifecycle of cluster and applications. It's modern operational framework which allow managing our resources using `Git` repositories as the single source of truth. In the platform, GitOps ensures that all changes are version-controlled, auditable, and easily reversible. By leveraging Git’s robust workflows, teams can automate deployments and rollbacks while maintaining consistency and reliability across environments. This capability empowers developers to implement continuous delivery practices, facilitating faster and more secure releases.

Reviewer 3: ```suggestion
Giant Swarm uses GitOps as a first-class citizen to manage the lifecycle of clusters and applications. It's a modern operational framework which allows for managing our resources using `Git` repositories as the single source of truth. In the platform, GitOps ensures that all changes are version-controlled, auditable, and easily reversible. By leveraging Git’s robust workflows, teams can automate deployments and rollbacks while maintaining consistency and reliability across environments. This capability empowers developers to implement continuous delivery practices, facilitating faster and more secure releases.
```

Reviewer 2: I think this is way too long of a description, but doesn't explain how GS "uses GitOps" – we don't enforce it, so the potential customer should be clear here what to expect. I'd rather describe GitOps as built-in capability instead of explaining the concept at length. I'm thinking of something like this: "[... explanation what GitOps is in one sentence ...] We pre-install Flux and you can use it. Only connect your Git control repo and you're good to go." (must be reworded, of course).

[Learn more in our guide about GitOps]({{< relref "/tutorials/continuous-deployment" >}}).

## Developer portal

> Giant Swarm uses [Backstage](https://www.cncf.io/projects/backstage/) as developer portal serving as the central hub for accessing all platform services and resources. It provides an intuitive interface where platform engineers and developers find documentation, manage their projects, and access tools necessary for application development and deployment. By consolidating resources and simplifying access, you developer portal enhances the user experience, reduces onboarding time, and fosters collaboration among teams. It acts as the gateway to the platform’s capabilities, streamlining workflows and promoting productivity.

Reviewer 3: ```suggestion
Giant Swarm uses [Backstage](https://www.cncf.io/projects/backstage/) as developer portal serving as the central hub for accessing all platform services and resources. It provides an intuitive interface where platform engineers and developers find documentation, manage their projects, and access tools necessary for application development and deployment. By consolidating resources and simplifying access, your developer portal enhances the user experience, reduces onboarding time, and fosters collaboration among teams. It acts as the gateway to the platform’s capabilities, streamlining workflows and promoting productivity.
```

Reviewer 2: We're not installing Backstage on customer MCs yet, right? So can we even mention it as such?

Author: We are on it, not sure what is the estimation, but I will ask honeybadger, I think it is important to include it

## Observability

> Observability on the Giant Swarm platform is based in Mimir, Loki and Grafana. The stack runs in the management cluster and provides a centralized view of the platform metrics, logs, and traces. You can add application metrics and logs letting the your teams detect anomalies, troubleshoot issues, and optimize resources efficiently. The platform supports customizable dashboards and alerts, enabling proactive management and a deeper understanding of application behavior and infrastructure status.

Reviewer 3: ```suggestion
Observability on the Giant Swarm platform is based on Mimir, Loki and Grafana. The stack runs in the management cluster and provides a centralized view of the platform metrics, logs, and traces. You can collect application metrics and logs letting your teams detect anomalies, troubleshoot issues, and optimize resources efficiently. The platform supports customizable dashboards and alerts as well as SLO-based monitoring, enabling proactive management and a deeper understanding of application behavior and infrastructure status.
```

Reviewer 2: ```suggestion
Observability on the Giant Swarm platform is based on the Grafana stack. Collectors store metrics, logs and traces in central storage and you can use them in the managed Grafana instance on your management cluster to get a centralized view, set up alerts and dashboards, or troubleshoot issues. In our tutorial (TODO LINK HERE OR BELOW), you see how easily your developers can start monitoring applications with this stack. Giant Swarm's pre-installed dashboards let you view the infrastructure and cluster health without extra effort.
```

> ## Single sign-On

Reviewer 2: ```suggestion
## Single sign-on
```

> Single Sign-On (SSO) simplifies user authentication across the Giant Swarm platform by allowing users to access multiple services with your existing identity provider. It enhances security and user convenience by reducing password fatigue and centralizing identity management. You can build on top of `Kubernetes` Role-Based Access Control (RBAC) to define fine-grained access policies and permissions, ensuring that only authorized users can interact with the platform resources.

Reviewer 2: ```suggestion
Single Sign-On (SSO) simplifies user authentication across your use of the Giant Swarm platform by authenticating user access with your existing identity provider (for example Google Workspaces or Microsoft Entra / Active Directory). This enhances security and user convenience by centralizing identity management and not requiring a separate password. You can build on top of Kubernetes Role-Based Access Control (RBAC) to define fine-grained access policies and permissions, ensuring which users are authorized to access certain platform resources.

TODO LINK OVERVIEW ARTICLE
```

## Network and connectivity

> The different technologies used in the Giant Swarm platform provides you with a secure and reliable network and connectivity between your applications and services. Platform engineers can create network policies to control traffic flow and enforce security rules, ensuring that data is protected and isolated. Relying on the built-in networking capabilities of `Kubernetes`, the platform supports many routing and load balancing strategies, enabling seamless communication between services and applications.

Reviewer 3: ```suggestion
The different technologies used in the Giant Swarm platform provide you with a secure and reliable network and connectivity between your applications and services. Platform engineers can create network policies to control traffic flows and enforce security rules, ensuring that data is protected and isolated. Relying on the built-in networking capabilities of `Kubernetes` as well as the extended capabilities of `Cilium CNI`, the platform supports many routing and load balancing strategies, enabling seamless communication between services and applications.
```

There's nothing about exposing services and ingress in here, I think that would need to be added, right? @pipo02mix

## Runtime security

> Runtime security focuses on protecting applications and infrastructure during execution. [Falco](https://www.cncf.io/projects/falco/) project provide you with intrusion detection, vulnerability scanning, and automated threat response capabilities. The Giant Swarm platform ensures that security policies are enforced consistently, reducing the attack surface and mitigating risks.

Reviewer 3: ```suggestion
Runtime security focuses on protecting applications and infrastructure during execution. [Falco](https://www.cncf.io/projects/falco/) provides you with intrusion detection, vulnerability scanning, and automated threat response capabilities. The Giant Swarm platform ensures that security policies are enforced consistently, reducing the attack surface and mitigating risks.
```

why are we only mentioning runtime security here? it's actually the one security capabaility our customers use least, they all use policivy management and most also CVE detection though

Author: it is coming from the chart creation. We removed the previous policy enforcement box. I could not remember why, do you @JosephSalisbury ?

Reviewer: i don't remember a particular reason - i think it's reasonable to add policy management, maybe roll CVE detection in with runtime scanning somehow

## Cost management

> Cost control is becoming increasingly important as organizations seek to optimize their cloud spending. Based on lessons learned from our experience with customers, the platform is built to help you manage costs effectively. Our solution engineers will help you to add visibility and control over your resource usage and expenses, leading to better decision-making and cost savings.

Reviewer 3: ```suggestion
Cost control is becoming increasingly important as organizations seek to optimize their cloud spending. Based on lessons learned from our experience with customers, the platform is built to help you manage costs effectively. Our solution engineers will help you add visibility and control over your resource usage and expenses, leading to better decision-making and cost savings. By working on resource optimization and auto-scaling as well as resource projections and reservations, we can lower costs, without impacting every developer individually.
```

added the work we do in optimizing that goes beyond just visibility and control

Reviewer 2: IMO still quite vague, or too long of a paragraph. Can we mention very explicitly what we do in this area? Auto-scaling and choice of instance types is one topic, but what else? Should we mention that we can do optimization projects with customers? Otherwise, it may sound like there's existing tooling to do all these things.

## Automatic cluster management

> Leaning on the `Cluster API`, the Giant Swarm platform provides solid automation to manage the lifecycle of your clusters. On top, our team has developed additional logic cover certain aspects of the cluster lifecycle not included in the upstream project.

Reviewer 3: ```suggestion
Based on `Cluster API`, the Giant Swarm platform provides solid automation to manage the lifecycle of your clusters. On top of it, our team has developed additional logic to cover certain essential aspects of the cluster lifecycle not included in the upstream project.
```

Reviewer 2: "Management" is such a vague word that one could replace it with "juggling" and not have a less informational sentence 😆.

Also I disagree with mentioning 'stuff not included upstream' in such a way. Without explaining what capabilities we offer like that, it's 1) an implementation detail and 2) screams "vendor lock-in".

I'm adding a suggestion below, covering both paragraphs – since the second paragraph greatly lists the capabilities.

> Leaning on the `Cluster API`, the Giant Swarm platform provides solid automation to manage the lifecycle of your clusters. On top, our team has developed additional logic cover certain aspects of the cluster lifecycle not included in the upstream project.
> 
> Our system includes automated provisioning, scaling, and upgrading of clusters, reducing the operational burden on your team. The platform ensures clusters are always running optimally and securely, allowing developers to focus on building and deploying applications without worrying about underlying infrastructure management.

Reviewer 2: ```suggestion
Based on Cluster API and other operators, the Giant Swarm platform provides automated provisioning, scaling, (planned) upgrading and deletion of clusters, reducing the operational burden on your team. The platform ensures clusters are always running optimally and securely, allowing developers to focus on building and deploying applications without worrying about underlying infrastructure management. In addition, special capabilities such as private clusters and routing are possible, allowing you to integrate the platform in your existing networks.
```

> ## Cluster Scaling

Reviewer 2: ```suggestion
## Cluster scaling
```

(style guide says not to use title case)

One of the key features of the Giant Swarm platform is the ability to scale clusters automatically. It allows for dynamic adjustment of resources based on demand, supporting both horizontal and vertical scaling, ensuring applications have the necessary resources to handle varying workloads. By automating the scaling process, the platform helps maintain performance and reliability while optimizing resource utilization and cost.

## Cloud resources provisioning

> Most likely, you have already external cloud resources which could be managed by the platform. Thanks to [Crossplane](https://www.cncf.io/projects/crossplane/) project, you can include all your infrastructure, such as databases, queues or buckets, as part of the platform, relying on the same principles `Kubernetes` offers. Centralizing the management of all the infrastructure in a single place, helps you to reduce the operational overhead and to have a better control over your resources.

Reviewer 3: ```suggestion
Most likely, you have already external cloud resources which could be managed by the platform. Thanks to [Crossplane](https://www.cncf.io/projects/crossplane/) included in the management API, you can include all your infrastructure, such as databases, queues or buckets, as part of the platform, relying on the same principles `Kubernetes` offers. Centralizing the management of all the infrastructure in a single place, helps you reduce the operational overhead and have a better control over your resources.
```

Reviewer 2: Last sentence – should we rather hint at ease for developers / self-service? Otherwise, it remains vague what "better control" means.
